test_runner/xctestrun.py

```diff
@@ -155,11 +155,26 @@ def Run(self, device_id, sdk, derived_data_dir, startup_timeout_sec,
     Returns:
       A value of type runner_exit_codes.EXITCODE.
     """
-    logging.info('Running test-without-building with device %s', device_id)
+    run_envs = dict(os.environ)
+    testOutputsDir = run_envs['TEST_UNDECLARED_OUTPUTS_DIR'] #test.outputs
+    outputDir, _ = os.path.split(testOutputsDir)
+    #_, testTarget = os.path.split(outputDir)
+    resultBundleName = 'test'
+    resultBundlePath = os.path.join(outputDir, resultBundleName + '.xcresult')
+    resultBundlePathZip = os.path.join(outputDir, resultBundleName + '.xcresult.zip')
+    
+    logging.info('deleting previous test.xcresult if exits')
```

Comment:
Mind adding a comment about why this is necessary? Also, does this still work when a given test is ran multiple times?